packages/plugin-core/src/commands/NoteLookupCommand.ts

```diff
@@ -429,6 +430,57 @@ export class NoteLookupCommand extends BaseCommand<
     return { uri, node: nodeNew, resp };
   }
 
+  private async getVaultForNewNote({
+    fname,
+    picker,
+  }: {
+    fname: string;
+    picker: DendronQuickPickerV2;
+  }) {
+    const engine = getEngine();
+
+    const vaultsWithMatchingFile = new Set(
+      NoteUtils.getNotesByFname({ fname, notes: engine.notes }).map(
+        (n) => n.vault.fsPath
+      )
+    );
+
+    // Try to get the default vault value.
+    let vault: DVault | undefined = picker.vault
+      ? picker.vault
+      : PickerUtilsV2.getVaultForOpenEditor();
+
+    // If our current context does not have vault or if our current context vault
+    // already has a matching file name we want to ask the user for a different vault.
+    if (vault === undefined || vaultsWithMatchingFile.has(vault.fsPath)) {
+      // Available vaults are vaults that do not have the desired file name.
+      const availVaults = engine.vaults.filter(
+        (v) => !vaultsWithMatchingFile.has(v.fsPath)
+      );
+
+      if (availVaults.length > 1) {
+        const promptedVault = await PickerUtilsV2.promptVault(availVaults);
+        if (promptedVault === undefined) {
+          // User must have cancelled vault selection.
+          vault = undefined;
+        } else {
+          vault = promptedVault;
+        }
+      } else if (availVaults.length === 1) {
+        // There is only a single vault that is available so we dont have to ask the user.
+        vault = availVaults[0];
+      } else {
+        // We should never reach this as "Create New" should not be available as option
+        // to the user when there are no available vaults.
+        ErrorFactory.createInvalidStateError({
```

Comment:
we are creating an error here but not returning it. should we log/return this error?

Reply:
Yep, that's a miss.

Since it should never occur and represents a bug, sounds like a good case to throw the error so we can be notified of the issue quicker.